examples/combined_ODNP.py

```diff
@@ -0,0 +1,352 @@
+tx_phases = r_[0.0,90.0,180.0,270.0]
+amplitude = 1.0
+nScans = 1
+nEchoes = 1
+date = datetime.now().strftime('%y%m%d')
+# all times in microseconds
+# note that acq_time_ms is always milliseconds
+p90_us = 4.464
+deadtime_us = 10.0
+repetition_us = 6e6
+FIR_rd = 3e6
+SW_kHz = 5 
+acq_time_ms = 500. # ms
+nPoints = int(acq_time_ms*SW_kHz+0.5)
+tau_adjust_us = 0.0
+deblank_us = 1.0
+tau_us = 3500
+pad_us = 0
+pul_prog = 'ODNP_v3'
+vd_list = r_[2.1e3,2.1e4,2.1e5,4.3e5,6.4e5,8.8e5,1.1e6,1.3e6,1.5e6,1.7e6,1.9e6,2.5e6,3e6,3.3e6]
+T1_powers_dB = r_[27,30,32,33]
+T1_node_names = ['FIR_27dBm','FIR_30dBm','FIR_32dBm','FIR_33dBm']
+
+# }}}
+#{{{Power settings
+max_power = 3 #W
+power_steps = 14
+threedown = True
+dB_settings = gen_powerlist(max_power,power_steps+1, threedown=threedown)
+print("dB_settings",dB_settings)
+print("correspond to powers in Watts",10**(dB_settings/10.-3))
+myinput = input("Look ok?")
+#}}}
+time_list = [time.time()]
+if myinput.lower().startswith('n'):
+    raise ValueError("you said no!!!")
+powers = 1e-3*10**(dB_settings/10.)
+time_list.append(time.time())
+#{{{ enhancement curve pulse prog
+def run_scans(nScans, power_idx, DNP_data=None):
+    """run nScans and slot them into the power_idx index of DNP_data -- assume
+    that the first time this is run, it will be run with DNP_data=None and that
+    after that, you will pass in DNP_data
+    
+    assume that the power axis is 1 longer than the "powers" array,
+    so that we can also store the thermally polarized signal in this array
+    (note that powers and other parameters are defined globally w/in the
+    script, as this function is not designed to be moved outside the module
+
+    this generates an "indirect" axis
+    """
+    print("about to run run_scans for",power_idx)
+    ph1_cyc = r_[0,1,2,3]
+    ph2_cyc = r_[0]
+    nPhaseSteps = len(ph1_cyc)*len(ph2_cyc)
+    total_pts = nPoints*nPhaseSteps
+    data_length = 2*nPoints*nEchoes*nPhaseSteps
+    for x in range(nScans):
+        run_scans_time_list = [time.time()]
+        run_scans_names = ['configure']
+        print("*** *** *** SCAN NO. %d *** *** ***"%(x+1))
+        print("\n*** *** ***\n")
+        print("CONFIGURING TRANSMITTER...")
+        SpinCore_pp.configureTX(adcOffset, carrierFreq_MHz, tx_phases, amplitude, nPoints)
+        print("\nTRANSMITTER CONFIGURED.")
+        run_scans_time_list.append(time.time())
+        run_scans_names.append('configure Rx')
+        print("***")
+        print("CONFIGURING RECEIVER...")
+        acq_time_ms = SpinCore_pp.configureRX(SW_kHz, nPoints, 1, nEchoes, nPhaseSteps)
+        # acq_time_ms is in msec!
+        print("ACQUISITION TIME IS",acq_time_ms,"ms")
+        verifyParams(nPoints=nPoints, nScans=nScans, p90_us=p90_us, tau_us=tau_us)
+        run_scans_time_list.append(time.time())
+        run_scans_names.append('init')
+        print("\nRECEIVER CONFIGURED.")
+        print("***")
+        print("\nINITIALIZING PROG BOARD...\n")
```

Comment:
do you actually read and use this?  scan number might be useful, but everything else that's noise should go to the log file, I would think.